pkg/kubelet/metrics/metrics.go

```diff
@@ -101,6 +101,14 @@ var (
 			Help:      "Interval in microseconds between relisting in PLEG.",
 		},
 	)
+	ProberResults = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: KubeletSubsystem,
+			Name:      "probe_result",
+			Help:      "The result of a liveness or readiness probe for a container.",
+		},
+		[]string{"probe_type", "container_name"},
```

Comment:
Hmm, I should have thought of this before, but I don't think metric per-container is viable. See https://prometheus.io/docs/practices/instrumentation/#do-not-overuse-labels. Each combination of labels becomes an separate timeseries.

At the very least, metrics will need to be cleaned up with [GaugeVec.Delete](https://godoc.org/github.com/prometheus/client_golang/prometheus#GaugeVec.Delete) when the container is deleted.

Reply:
Any alternatives you can recommend? I can add GaugeVec.Delete in the meantime.

Reply:
Added GaugeVec.Delete. PTAL